pkg/sshkey/clusterkey.go

```diff
@@ -0,0 +1,76 @@
+package sshkey
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/cnrancher/autok3s/pkg/common"
+	"github.com/cnrancher/autok3s/pkg/types"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	PrivateKeyFilename  = "id_rsa"
+	PublicKeyFilename   = "id_rsa.pub"
+	CertificateFilename = "pub.cert"
+)
+
+func StoreClusterSSHKeys(clusterName string, ssh *types.SSH) (*types.SSH, error) {
+	var rtn *types.SSH
+	if !NeedSSHKeys(*ssh) {
+		return nil, nil
+	}
+	base := common.GetClusterContextPath(clusterName)
+	// copy key from stored ssh key pair
+	if ssh.SSHKeyName != "" && ssh.SSHCertPath == "" && ssh.SSHKeyPath == "" {
+		keys, err := common.DefaultDB.ListSSHKey(&ssh.SSHKeyName)
+		if err != nil {
+			return nil, err
+		}
+		if len(keys) != 1 {
+			return nil, fmt.Errorf("failed to get ssh key %s from db", ssh.SSHKeyName)
+		}
+		key := keys[0]
+		rtn = ssh
+		rtn.SSHKey = key.SSHKey
+		rtn.SSHCert = key.SSHCert
+	}
+
+	// save cert to cert path
+	if ssh.SSHCertPath == "" && ssh.SSHCert != "" {
+		if rtn == nil {
+			rtn = ssh
+		}
+		certPath := filepath.Join(base, CertificateFilename)
+		if err := os.RemoveAll(certPath); err != nil {
+			return nil, err
+		}
+		if err := ioutil.WriteFile(certPath, []byte(ssh.SSHCert), 0644); err != nil {
+			return nil, errors.Wrapf(err, "failed to write cluster ssh cert to file %s", certPath)
+		}
+		rtn.SSHCertPath = certPath
+	}
+
+	// save key to key path
+	if ssh.SSHKeyPath == "" && ssh.SSHKey != "" {
+		if rtn == nil {
+			rtn = ssh
+		}
+		keyPath := filepath.Join(base, PrivateKeyFilename)
```

Comment:
We will replace the local private key to the key path every time when we use `--ssh-key-name`?

Reply:
No, this code should be only run once when initializing the cluster first time with `--ssh-key-name`. The `os.RemoveAll` is just to remove the current key file just in case in the cluster context folder which now is `<cfgPath>/<cluster-context-name>`